impl/src/planning.rs

```diff
@@ -355,20 +364,72 @@ impl<'fetcher> BuildPlanner for BuildPlannerImpl<'fetcher> {
   fn plan_build(
     &mut self,
     settings: &RazeSettings,
+    path_prefix: &PathBuf,
     files: CargoWorkspaceFiles,
     platform_details: Option<PlatformDetails>,
   ) -> Result<PlannedBuild> {
-    let metadata = self.metadata_fetcher.fetch_metadata(&files)?;
+    let metadata = self.metadata_fetcher.fetch_metadata(&files, true)?;
+
+    // Create one combined metadata object which includes all dependencies and binaries
     let crate_catalog = CrateCatalog::new(&metadata)?;
 
-    // Generate additional PlatformDetails
+    // Additionally, fetch metadata for the list of binaries present in raze settings. This
```

Comment:
Could this bin crates work be done in a helper function, to help keep this function small enough to stay in reader's brain?

Reply:
Updated, let me know what you think.